### ISSUE TITLE
Allow creating more than one type of compute node instance

### ISSUE THREAD

**Issue by cmd-ntrf** [ComputeCanada/magic_castle#29]
To achieve need, we need to first build a list of hashes that contain the type of the instance and a unique name.

Here is a first draft of how to build the list of hashes.
```
locals {
    nodes = [
        { type = "n1-standard-2", count = 2 }, 
        { type = "n1-standard-4", count = 3 }
    ]
    gpu_per_node = [ 
        { type = "nvidia-tesla-k80", count = 0 },
        { type = "nvidia-tesla-k80", count = 1 },
    ]
    compute = flatten([
        for i in range(length(local.nodes)): [
            for j in range(local.nodes[i].count): { 
                type = local.nodes[i].type, 
                # name = "${replace(replace(local.nodes[i].type, "_", ""), "-", "")}-node-${j+1}"
                name = "${local.nodes[i].type}-node-${j+1}"
                gpu_type = local.gpu_per_node[i].type
                gpu_count = local.gpu_per_node[i].count
            }
        ]
    ])
}
output "compute" {
    value = local.compute
}

```

**Comment by cmd-ntrf** [comment]
Work in progress in branch `heteronode`. So far, openstack, ovh and gcp support a list of node type and count.

**Comment by cmd-ntrf** [comment]
Fix by #59.
